From 8a18efab6eb6f0bfe220ba3644d438fd431bd529 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 20 Oct 2016 16:58:43 -0700
Subject: [PATCH] TOOLS-1584 imgapi client listImages() 'filters.channel' arg
 should win over 'this.channel' Reviewed by: Julien Gilli
 <julien.gilli@joyent.com> Approved by: Julien Gilli <julien.gilli@joyent.com>

---
 lib/imgapi.js | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/lib/imgapi.js b/lib/imgapi.js
index e603170..4dbfdd7 100644
--- a/lib/imgapi.js
+++ b/lib/imgapi.js
@@ -563,10 +563,10 @@ IMGAPI.prototype.listImages = function listImages(filters, options, callback) {
 
     function listImagesWithLimit(headers, cb) {
         // limit and marker come straight from filters
-        var path = self._path('/images', filters, {
+        var path = self._path('/images', {
             channel: self.channel,
             inclAdminFields: inclAdminFields
-        });
+        }, filters);
         var reqOpts = {
             path: path,
             headers: headers
@@ -609,10 +609,10 @@ IMGAPI.prototype.listImages = function listImages(filters, options, callback) {
                 filters.limit = limit;
             }
 
-            var path = self._path('/images', filters, {
+            var path = self._path('/images', {
                 channel: self.channel,
                 inclAdminFields: inclAdminFields
-            });
+            }, filters);
             var reqOpts = {
                 path: path,
                 headers: headers
-- 
2.21.0

